public/app/plugins/datasource/prometheus/components/PromQueryField.tsx

```diff
@@ -163,9 +163,10 @@ class PromQueryField extends React.PureComponent<PromQueryFieldProps, PromQueryF
     const {
       data,
       datasource: { languageProvider },
+      range,
     } = this.props;
 
-    if (languageProvider !== prevProps.datasource.languageProvider) {
+    if (languageProvider !== prevProps.datasource.languageProvider || !_.isEqual(range, prevProps.range)) {
```

Comment:
One last thought - here instead of comparing the whole moment object, we could just compare the absolute ranges

```
const absoluteRange: AbsoluteTimeRange = { from: range.from.valueOf(), to: range.to.valueOf() };
const prevAbsoluteRange: AbsoluteTimeRange = { from: prevProps.range.from.valueOf(), to: prevProps.range.to.valueOf() };
```

and replace it with `!_.isEqual(absoluteRange, prevAbsoluteRange))`